staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -319,9 +319,6 @@ type Config struct {
 	// number of active watch request(s) in flight and during shutdown
 	// it will wait, at most, for the specified duration and allow these
 	// active watch requests to drain with some rate limiting in effect.
-	// The default is zero, which implies the apiserver will not keep
```

Comment:
is the cost of keeping track of pending watch requests meaningful? I want to make sure we're not defaulting configs into something noticeably more expensive for kube-apiserver to run

Reply:
maybe @wojtek-t knows

Reply:
If you ask about resource consumption (cpu/mem), it's fairly cheap. We basically only have a rate-limited waiting group:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/waitgroup/ratelimited_waitgroup.go

and closing watches is rate-limited when apiserver is in shutdown:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/waitgroup/ratelimited_waitgroup.go#L70

Corresponding PR adding this logic: https://github.com/kubernetes/kubernetes/pull/114925/files

So I wouldn't worry about the resource overhead.

What changes though is that apiserver graceful shutdown can now be visibly longer (by this 67s) by default with this change. I think this is fine, but we need to weigh that in.

Reply:
this also need to be updated 

https://github.com/kubernetes/kubernetes/blob/c90a4b16b6aa849ed362ee40997327db09e3a62d/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L292-L297